Issue #4280

ankitkataria opened the pull request:
Hey, 
I was trying to solve issue 4280
- I added the required page_white_put icon
- i realised that the themes of OBS is managed in a separate repo and update_gento.sh copied all the required assets(images and icons) from the project, so i added a line to copy the icon, and ran the script. - I made changes to icons.scss and _files_view.html.erb. **BUT** the icon doesn't still seem to work i still get the old icon. 
- I figured that OBS is using image spriting, i tried reading the file /src/api/lib/tasks/sprites.rake, i figured this is the file that generates the sprited image, but i couldn't figure out when it is called, and could i call it to make a new sprited image with the newly added icon.
Any help?

ankitkataria commented:
@Ana06 Any help?

bgeuken commented:
Hi @ankitkataria, first of all thanks for your contribution:bouquet:

It looks like you did not update the [sprite file](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/assets/images/icons_sprite.png). Just run the script;-)
This will also update the [icons.scss](assets/stylesheets/webui/application/icons.scss) properly.

bgeuken commented on src/api/app/assets/stylesheets/webui/application/icons.scss:
```diff
@@ -120,6 +120,7 @@
 .icons-page_white_add { width: 16px; height: 16px; background: image-url('icons_sprite.png') -342px -400px no-repeat; }
 .icons-page_white_delete { width: 16px; height: 16px; background: image-url('icons_sprite.png') -362px -400px no-repeat; }
 .icons-page_white_get { width: 16px; height: 16px; background: image-url('icons_sprite.png') -382px -400px no-repeat; }
+.icons-page_white_put { width: 16px; height: 16px; background: image-url('icons_sprite.png') -382px -400px no-repeat; }
```
Note the px coordinates (eg. ` -382px -400px `) at the end of each line. Each of them points to a single icon on the [sprite file](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/assets/images/icons_sprite.png).

bgeuken commented:
@ankitkataria and please add a short explanation to the commit message why you add the page_white_get icon.

Thanks:-)

ankitkataria commented:
![image](https://user-images.githubusercontent.com/22021150/35435944-89a19db6-02b2-11e8-824a-398b940d8599.png)
fixed the download icon, what all i did:
- changed update_bento.sh to also copy the page_white_put icon from [opensuse themes](https://github.com/openSUSE/opensuse-themes) to the required assets/icons/ directory
- changed _files_view.html.erb 
- ran the assets:reprite task to produce the newly sprited images and update icons.scss

thankyou both for the help and support. :smile:

bgeuken commented:
@ankitkataria Regarding the comment, what @Ana06 meant was to add this to the bottom of the commit:

`
Fixes #4280
`

That way GitHub automatically closes the issue when the PR get's merged.

ankitkataria commented:
thankyou @bgeuken , i will take take of the commit messages from the next time. :smile:

Ana06 commented:
@bgeuken 

> @ankitkataria Regarding the comment, what @Ana06 meant was to add this to the bottom of the commit:
> 
> Fixes #4280
> 
> That way GitHub automatically closes the issue when the PR get's merged.

Yes! Do not add the issue to the title 🙈 The title should be something short that allow you know what the commit is about, and nobody remember issues numbers 😉 So, just wiriting `Fixes #4280` at the end of the description is enough. 👍

bgeuken commented:
@ankitkataria @Ana06 I noticed, but one thing at a time;-)
